src/webui/api/torrentscontroller.h

```diff
@@ -28,15 +28,30 @@
 
 #pragma once
 
+#include <QHash>
+#include <QSet>
+
 #include "apicontroller.h"
+#include "base/bittorrent/torrentdescriptor.h"
+
+namespace BitTorrent
+{
+    class InfoHash;
+    class TorrentInfo;
+}
+
+namespace Net
+{
+    class DownloadResult;
```

Comment:
```suggestion
    struct DownloadResult;
```

Reply:
Aha! That explains why `TorrentDescriptor` couldn't be forward declared and had to be imported in `torrentscontroller.h`.